src/sentry/integrations/opsgenie/integration.py

```diff
@@ -180,50 +182,48 @@ def update_organization_config(self, data: MutableMapping[str, Any]) -> None:
-            if (team["team"], team["integration_key"]) in existing_team_key_pairs:
-                continue
-
-            integration_key = team["integration_key"]
-
-            # validate integration keys
-            client = OpsgenieClient(
-                integration=integration,
-                integration_key=integration_key,
-            )
-            # call an API to test the integration key
-            try:
-                client.get_alerts()
-            except ApiError as e:
-                logger.info(
-                    "opsgenie.authorization_error",
-                    extra={"error": str(e), "status_code": e.code},
+        with record_event(OnCallInteractionType.VERIFY_KEYS).capture():
+            for team in teams:
+                # skip if team, key pair already exist in config
+                if (team["team"], team["integration_key"]) in existing_team_key_pairs:
+                    continue
+
+                integration_key = team["integration_key"]
+
+                # validate integration keys
+                client = OpsgenieClient(
+                    integration=integration,
+                    integration_key=integration_key,
                 )
-                if e.code == 429:
-                    raise ApiRateLimitedError(
-                        "Too many requests. Please try updating one team/key at a time."
-                    )
-                elif e.code == 401:
-                    invalid_keys.append(integration_key)
-                    pass
-                elif e.json and e.json.get("message"):
-                    raise ApiError(e.json["message"])
-                else:
-                    raise
-
-        if invalid_keys:
-            raise ApiUnauthorized(f"Invalid integration key: {str(invalid_keys)}")
+                # call an API to test the integration key
+                try:
+                    client.get_alerts()
+                except ApiError as e:
+                    if e.code == 429:
+                        raise ApiRateLimitedError(
+                            "Too many requests. Please try updating one team/key at a time."
+                        )
+                    elif e.code == 401:
+                        invalid_keys.append(integration_key)
+                        pass
+                    elif e.json and e.json.get("message"):
+                        raise ApiError(e.json["message"])
+                    else:
+                        raise
+
+            if invalid_keys:
+                raise ApiUnauthorized(f"Invalid integration key: {str(invalid_keys)}")
 
         return super().update_organization_config(data)
 
     def schedule_migrate_opsgenie_plugin(self):
-        migrate_opsgenie_plugin.apply_async(
-            kwargs={
-                "integration_id": self.model.id,
-                "organization_id": self.organization_id,
-            }
-        )
+        with record_event(OnCallInteractionType.MIGRATE_PLUGIN).capture():
```

Comment:
capturing this doesn't really give us any insight as this is only queuing up the task. i think you want to capture the `migrate_opsgenie_plugin` function instead